ouroboros-consensus-cardano/test/cardano-test/Test/Consensus/Cardano/Serialisation.hs

```diff
@@ -77,3 +88,46 @@ prop_CardanoBinaryBlockInfo blk =
     encodedNestedHeader :: Lazy.ByteString
     encodedNestedHeader = case encodeDepPair testCodecCfg (unnest (getHeader blk)) of
         GenDepPair _ (Serialised bytes) -> bytes
+
+{-------------------------------------------------------------------------------
+  Test that our encoders produce valid CBOR
+-------------------------------------------------------------------------------}
+
+encodersProduceValidCBOR ::
+     forall blk . (SerialiseDiskConstraints blk)
+  => CodecConfig blk
+  -> Examples blk
+  -> TestTree
```

Comment:
Maybe this function could go into the consensus testlib

Reply:
Yeah, then it could also be re-used in `Test.Consensus.{Byron,Shelley}.Serialisation` :+1: